database/engine/metadata_log/metadatalogapi.c

```diff
@@ -221,16 +235,108 @@ void metalog_commit_delete_dimension(RRDDIM *rd)
         return;
 
     ctx = host->rrdeng_ctx->metalog_ctx;
+    if (!ctx) /* metadata log has not been initialized yet */
+        return;
     buffer = buffer_create(64); /* This will be freed after it has been committed to the metadata log buffer */
 
-    buffer_sprintf(buffer, "CONTEXT %s\n", st->id);
-
-    uuid_unparse_lower(*rd->state->rrdeng_uuid, uuid_str);
+    uuid_unparse_lower(*rd->state->metric_uuid, uuid_str);
     buffer_sprintf(buffer, "TOMBSTONE %s\n", uuid_str);
 
     metalog_commit_creation_record(ctx, buffer);
 }
 
+RRDSET *metalog_get_chart_from_uuid(struct metalog_instance *ctx, uuid_t *chart_uuid)
+{
+    GUID_TYPE ret;
+    char chart_object[33], chart_fullid[RRD_ID_LENGTH_MAX + 1];
+    uuid_t *machine_guid, *chart_char_guid;
+
+    ret = find_object_by_guid(chart_uuid, chart_object, 33);
+    assert(GUID_TYPE_CHART == ret);
+
+    machine_guid = (uuid_t *)chart_object;
+    RRDHOST *host = ctx->rrdeng_ctx->host;
+    assert(!uuid_compare(host->host_uuid, *machine_guid));
+
+    chart_char_guid = (uuid_t *)(chart_object + 16);
+
+    ret = find_object_by_guid(chart_char_guid, chart_fullid, RRD_ID_LENGTH_MAX + 1);
+    assert(GUID_TYPE_CHAR == ret);
+    RRDSET *st = rrdset_find(host, chart_fullid);
+
+    return st;
+}
+
+RRDDIM *metalog_get_dimension_from_uuid(struct metalog_instance *ctx, uuid_t *metric_uuid)
+{
+    GUID_TYPE ret;
+    char dim_object[49], chart_object[33], id_str[1024], chart_fullid[RRD_ID_LENGTH_MAX + 1];
+    uuid_t *machine_guid, *chart_guid, *chart_char_guid, *dim_char_guid;
+
+    ret = find_object_by_guid(metric_uuid, dim_object, 49);
+    if (GUID_TYPE_DIMENSION != ret) /* not found */
+        return NULL;
+
+    machine_guid = (uuid_t *)dim_object;
+    RRDHOST *host = ctx->rrdeng_ctx->host;
+    assert(!uuid_compare(host->host_uuid, *machine_guid));
+
+    chart_guid = (uuid_t *)(dim_object + 16);
+    dim_char_guid = (uuid_t *)(dim_object + 16 + 16);
+
+    ret = find_object_by_guid(dim_char_guid, id_str, 1024);
+    assert(GUID_TYPE_CHAR == ret);
+
+    ret = find_object_by_guid(chart_guid, chart_object, 33);
```

Comment:
sizeof

Reply:
Yeah, we've got A LOT of magic numbers in this branch for some reason, I'm afraid this is not the only occurrence.